ext/fetch/lib.rs

```diff
@@ -1082,11 +1094,41 @@ type Connector = proxy::ProxyConnector<HttpConnector>;
 #[allow(clippy::declare_interior_mutable_const)]
 const STAR_STAR: HeaderValue = HeaderValue::from_static("*/*");
 
+#[derive(Debug)]
+pub struct ClientSendError {
+  uri: Uri,
+  source: hyper_util::client::legacy::Error,
+}
+
+impl ClientSendError {
+  pub fn is_connect_error(&self) -> bool {
+    self.source.is_connect()
+  }
+}
+
+impl std::fmt::Display for ClientSendError {
+  fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+    write!(
+      f,
+      "error sending request for url ({uri}): {source}",
+      uri = self.uri,
+      // NOTE: we can use `std::error::Report` instead once it's stabilized.
+      source = error_reporter::Report::new(&self.source),
+    )
+  }
+}
+
+impl std::error::Error for ClientSendError {
+  fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+    Some(&self.source)
+  }
+}
```

Comment:
For reviewers: this custom error type is needed because we need to preserve the original error so we can later access with `std::error::Error::source` to get `hyper::Error`, while having to adjust the error message format using `error_reporter::Report`.